src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+
+    private int size = 0;
+    private Object[] array = new Object[ARRAY_SIZE];
+
     @Override
     public void add(T value) {
-
+        if (size + 1 == array.length) {
+            increaseArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index out of bounds");
+        }
+        if (size == index) {
+            add(value);
+            return;
+        }
+        if (size + 1 == array.length) {
+            increaseArray();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size <= index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index out of bounds");
+        }
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index out of bounds");
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index out of bounds");
+        }
+        T removedElement = (T) array[index];
+        size--;
+        System.arraycopy(array, index + 1, array, index, size - index);
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == element || (array[i] != null && array[i].equals(element))) {
+                size--;
+                System.arraycopy(array, i + 1, array, i, size - i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("element not exist in list");
```

Comment:
better find the index of that element in the array and use remove(index), if the element is missing, throw your exception